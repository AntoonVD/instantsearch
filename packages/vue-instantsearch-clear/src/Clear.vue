<template>
  <button type="reset"
          class="alg-clear-search"
          :class="{'alg-clear-search--disabled': disabled}"
          :disabled="disabled"
          @click.prevent="clear"
  >
    <slot>
      <span class="alg-clear-search__label">Clear</span>
    </slot>
  </button>
</template>

<script>
  import algoliaComponent from 'vue-instantsearch-component'

  export default {
    mixins: [algoliaComponent],
    props: {
      clearQuery: {
        type: Boolean,
        required: false,
        default: true
      },
      clearFacets: {
        type: Boolean,
        required: false,
        default: true
      }
    },
    computed: {
      disabled: function () {
        if (this.clearQuery && this.searchStore.query.length > 0) {
          return false
        }

        if (this.clearFacets && this.searchStore.activeRefinements.length > 0) {
          return false
        }

        return true
      }
    },
    methods: {
      clear: function () {
        this.searchStore.stop()
        if (this.clearQuery && this.searchStore.query.length > 0) {
          this.searchStore.query = ''
        }

        if (this.clearFacets && this.searchStore.activeRefinements.length > 0) {
          this.searchStore.clearRefinements()
        }
        this.searchStore.start()
      }
    }
  }
</script>

