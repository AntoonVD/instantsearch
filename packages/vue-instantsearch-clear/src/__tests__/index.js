import Clear from 'vue-instantsearch-clear';
import Vue from 'vue';

describe('Clear component', () => {
  const Component = Vue.extend(Clear);

  const stop = jest.fn();
  const start = jest.fn();
  const clearRefinements = jest.fn();

  const searchStore = {
    query: 'whatever',
    activeRefinements: ['whatever'],
    stop,
    start,
    clearRefinements
  };

  beforeEach(() => {
    stop.mockClear();
    start.mockClear();
    clearRefinements.mockClear();
    searchStore.query = 'whatever';
    searchStore.activeRefinements = ['whatever'];
  });

  test('Can clear the search query and the facets at the same time', () => {
    const vm = new Component({
      propsData: { searchStore }
    });

    vm.clear();
    expect(searchStore.query).toEqual('');
    expect(stop).toHaveBeenCalledTimes(1);
    expect(clearRefinements).toHaveBeenCalledTimes(1);
    expect(start).toHaveBeenCalledTimes(1);
  });

  test('Can disable query clearing', () => {
    searchStore.query = 'whatever';
    const vm = new Component({
      propsData: { searchStore, clearQuery: false }
    });
    vm.clear();
    expect(searchStore.query).toEqual('whatever');
    expect(stop).toHaveBeenCalledTimes(1);
    expect(clearRefinements).toHaveBeenCalledTimes(1);
    expect(start).toHaveBeenCalledTimes(1);
  });

  test('Can disable facets clearing', () => {
    searchStore.query = 'whatever';
    const vm = new Component({
      propsData: { searchStore, clearFacets: false }
    });
    vm.clear();
    expect(searchStore.query).toEqual('');
    expect(stop).toHaveBeenCalledTimes(1);
    expect(clearRefinements).not.toHaveBeenCalled();
    expect(start).toHaveBeenCalledTimes(1);
  });

  test('Has proper HTML rendering', () => {
    const vm = new Component({
      propsData: { searchStore }
    });
    vm.$mount();

    expect(vm.$el.outerHTML).toMatchSnapshot();
  });

  test('Has proper HTML rendering when disabled', () => {
    searchStore.query = '';
    searchStore.activeRefinements = [];
    const vm = new Component({
      propsData: { searchStore }
    });
    vm.$mount();

    expect(vm.$el.outerHTML).toMatchSnapshot();
  });
});
