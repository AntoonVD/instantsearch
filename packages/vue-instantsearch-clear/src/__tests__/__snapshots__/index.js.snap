// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear component has proper HTML rendering 1`] = `

<button type="reset"
        class="ais-clear"
>
  <span class="ais-clear__label">
    Clear
  </span>
</button>

`;

exports[`Clear component has proper HTML rendering when disabled 1`] = `

<button type="reset"
        disabled="disabled"
        class="ais-clear ais-clear--disabled"
>
  <span class="ais-clear__label">
    Clear
  </span>
</button>

`;
